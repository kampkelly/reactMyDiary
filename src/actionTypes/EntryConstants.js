export const NEW_ENTRY = 'NEW_ENTRY';
