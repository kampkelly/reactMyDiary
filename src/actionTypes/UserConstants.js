export const SIGNIN = 'SIGNIN';
export const SIGNUP = 'SIGNUP';
export const SIGNOUT = 'SIGNOUT';
export const PROFILE = 'PROFILE';
export const NOTIFICATION = 'NOTIFICATION';