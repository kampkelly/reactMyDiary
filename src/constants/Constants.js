const BASE_URL = 'https://kampkelly-mydiary-api.herokuapp.com';
// const { BASE_URL } = process.env;
export const userConstant = {
  SIGNIN_URL: `${BASE_URL}/api/v1/auth/login`,
};
