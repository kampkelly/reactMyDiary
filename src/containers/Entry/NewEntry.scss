$primary-color: #052F60;
$secondary-color: #DFAC2C;
$light-white-color: #8c8c8c;
$white-color: #fafafa;
$smartphone: "(min-width: 0px) and (max-width: 599px)";
$tablet: "(min-width: 600px) and (max-width: 900px)";

main {
	// background-image: url("../images/2.jpg");
  background-size: cover;
	background-position: center;
	background-repeat-x: no-repeat;
  background-repeat-y: no-repeat;
}
.grid-container {
	&>div {
		margin: auto;
	}
}
#add {
    padding-top: 60px;
	.cancel-div {
	  padding-right: 0;
	}
	.form-cover {
        background: rgba(107, 101, 101, 0.3);
        margin: auto;
    }
    form {
        .field {
            padding: 10px;
            .control-form, button {
            }
        }
		button {
			background-color: $primary-color;
	        color: white;
			width: 100%;
	        &:hover {
				background-color: #0d529d;
				color: white;
				transition: 0.5s;
	        }
		}
    }
	@media #{$tablet} {
		[class*='col-'] {
		  padding-right: 0;
		}
    }
}
