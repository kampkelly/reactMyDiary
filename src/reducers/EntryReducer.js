import { NEW_ENTRY, SHOW_ENTRY } from '../actionTypes/EntryConstants';
import { asyncActionName } from '../util/AsyncUtil';


const initialState = {
  entries: [],
  entry: {},
  loading: false,
  message: '',
  success: false,
  failure: false,
};

const entryReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case asyncActionName(NEW_ENTRY).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(NEW_ENTRY).success:
      return {
        ...state,
        success: action.payload
      };
    case asyncActionName(NEW_ENTRY).failure:
      document.getElementById('loading').style.display = 'none';
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(SHOW_ENTRY).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SHOW_ENTRY).success:
      return {
        ...state,
        entry: action.payload, success: true
      };
    case asyncActionName(SHOW_ENTRY).failure:
      return {
        ...state, message: action.payload.error
      };
    default:
      return state;
  }
};

export default entryReducer;