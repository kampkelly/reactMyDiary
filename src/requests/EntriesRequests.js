import axios from 'axios';

import { asyncActions } from '../util/AsyncUtil';
import { ALL_ENTRIES } from '../actionTypes/EntryConstants';
import { entryConstant } from '../constants/Constants';

export const PaginatedEntries = limit => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}?limit=${limit}`)
    .then((response) => {
      dispatch(asyncActions(ALL_ENTRIES).loading(false));
      dispatch(asyncActions(ALL_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(ALL_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(ALL_ENTRIES)
      .failure(true, error.response.data.message)));
};

export const AllEntries = () => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}`)
    .then((response) => {
      dispatch(asyncActions(ALL_ENTRIES).loading(false));
      dispatch(asyncActions(ALL_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(ALL_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(ALL_ENTRIES)
      .failure(true, error.response.data.message)));
};
