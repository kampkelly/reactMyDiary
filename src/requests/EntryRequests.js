import axios from 'axios';
import { createBrowserHistory } from 'history';

import { asyncActions } from '../util/AsyncUtil';
import { NEW_ENTRY, SHOW_ENTRY, UPDATE_ENTRY, ALL_ENTRIES, PAGINATED_ENTRIES, DELETE_ENTRY } from '../actionTypes/EntryConstants';
import { entryConstant } from '../constants/Constants';

const history = createBrowserHistory({ forceRefresh: true });

export const AddEntry = (title, description) => (dispatch) => {
  axios.post(`${entryConstant.ENTRIES_URL}`, { title, description })
    .then((response) => {
      dispatch(asyncActions(NEW_ENTRY).loading(false));
      dispatch(asyncActions(NEW_ENTRY).success(true));
      if (response.data.status === 'Success') {
        document.getElementById('loading').style.display = 'none';
        history.push(`/entries/${response.data.entry.id}?notice=${response.data.message}`);
      } else {
        document.getElementById('loading').style.display = 'none';
        document.querySelector('.form_error_text').style.display = 'block';
        document.querySelector('.form_error_text small').textContent = response.data.message;
      }
      dispatch(asyncActions(NEW_ENTRY).loading(true));
    })
    .catch(error => dispatch(asyncActions(NEW_ENTRY)
      .failure(true, error.response.data.message)));
};

export const ShowEntry = id => (dispatch) => {
  // document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src="http://res.cloudinary.com/ddfepbdqg/image/upload/v1543597369/Rolling.svg" id="loading"></img>`)
  axios.get(`${entryConstant.ENTRIES_URL}/${id}`)
    .then((response) => {
      dispatch(asyncActions(SHOW_ENTRY).loading(false));
      document.getElementById('loading').style.display = 'none';
      if (response.data.status === 'Success') {
        dispatch(asyncActions(SHOW_ENTRY).success(response.data.entry));
      }
      dispatch(asyncActions(SHOW_ENTRY).loading(true));
    })
    .catch(error => dispatch(asyncActions(SHOW_ENTRY)
      .failure(true, error.response.data.message)));
};

export const ModifyEntry = (id, title, description) => (dispatch) => {
  axios.put(`${entryConstant.ENTRIES_URL}/${id}`, { title, description })
    .then((response) => {
      dispatch(asyncActions(UPDATE_ENTRY).loading(false));
      if (response.data.status === 'Success') {
        dispatch(asyncActions(UPDATE_ENTRY).success(response.data.entry));
        document.getElementById('loading').style.display = 'none';
        history.push(`/entries/${response.data.entry.id}?notice=${response.data.message}`);
      } else {
        document.getElementById('loading').style.display = 'none';
        document.querySelector('.form_error_text').style.display = 'block';
        document.querySelector('.form_error_text small').textContent = response.data.message;
      }
      dispatch(asyncActions(UPDATE_ENTRY).loading(true));
    })
    .catch(error => dispatch(asyncActions(UPDATE_ENTRY)
      .failure(true, error.response.data.message)));
};

export const AllEntries = () => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}`)
    .then((response) => {
      document.getElementById('loading').style.display = 'none';
      dispatch(asyncActions(ALL_ENTRIES).loading(false));
      dispatch(asyncActions(ALL_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(ALL_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(ALL_ENTRIES)
      .failure(true, error.response.data.message)));
};

export const PaginatedEntries = limit => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}?limit=${limit}`)
    .then((response) => {
      dispatch(asyncActions(PAGINATED_ENTRIES).loading(false));
      dispatch(asyncActions(PAGINATED_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(PAGINATED_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(PAGINATED_ENTRIES)
      .failure(true, error.response.data.message)));
};

export const DeleteEntry = id => (dispatch) => {
  axios.delete(`${entryConstant.ENTRIES_URL}/${id}`)
    .then((response) => {
      dispatch(asyncActions(DELETE_ENTRY).loading(false));
      dispatch(asyncActions(DELETE_ENTRY).success(true));
      dispatch(asyncActions(DELETE_ENTRY).loading(true));
      history.push('/dashboard?notice=Entry has been deleted');
    })
    .catch(error => dispatch(asyncActions(DELETE_ENTRY)
      .failure(true, error.response.data.message)));
};