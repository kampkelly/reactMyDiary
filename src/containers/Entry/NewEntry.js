import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './NewEntry.scss';
import '../../styles/form.scss';
import '../../styles/sidebar.scss';
import { AddEntry } from '../../requests/EntryRequests';
import { validateForm } from '../../helpers/validateForm';

/**
 * @class NewEntry
 * @extends {Component}
 */
class NewEntry extends Component {
  /**
   *Creates an instance of NewEntry.
   * @memberof NewEntry
   */
  constructor() {
    super();
    this.state = {
      entry: {},
      title: '',
      description: '',
      success: false
    };
    this.addEntry = this.addEntry.bind(this);
    this.change = this.change.bind(this);
    this.goToHome = this.goToHome.bind(this);
  }

  /**
   *
   * @returns {jsx} - jsx
   * @memberof NewEntry
   */
  addEntry(e) {
    const errorMessage = 'One or more of the required fields are empty!';
    const values = { title: this.state.title, description: this.state.description };
    const noneEmpty = validateForm(errorMessage, values, e);
    if (noneEmpty === true) {
      document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src="http://res.cloudinary.com/ddfepbdqg/image/upload/v1543597369/Rolling.svg" id="loading"></img>`);
      this.props.AddEntry(this.state.title, this.state.description);
    }
  }

  change(e) {
    this.setState({ [e.target.name]: e.target.value })
  }

  goToHome() {
    this.props.history.push('/dashboard');
  }

  /**
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   * @memberof Header
   */
  render() {
    const { title, description } = this.state;
    return (
      <div>
        <section id="add">
          <div className="blue-forms form-cover text-center">
            <form className="add_form" id="add-form">
              <h3 className="title text-center white-text">What is happening today?</h3>
              <p className="danger-text form_error_text"><small></small></p>
              <p className="success-text form_success_text"><small></small></p>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <label className="white-text">Title:</label>
                </div>
                <div className="col-4-6 col-6-6-md col-6-6-xs">
                  <input type="text" required="true" className="control-form" id="title" name="title" value={title} onChange={this.change} autoComplete="title" />
                </div>
              </div>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <label className="white-text">Description:</label>
                </div>
                <div className="col-4-6 col-6-6-md col-6-6-xs">
                  <textarea className="control-form" id="description" required="true" name="description" value={description} onChange={this.change} resize="no"></textarea>
                </div>
              </div>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs"></div>
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <button type="submit" className="submit-button button-white" onClick={this.addEntry}>Save to diary</button>
                </div>
                <div className="col-2-6 col-6-6-md col-6-6-xs cancel-div">
                  <button type="submit" className="submit-button button-cancel mt-4-xs mt-4-md" onClick={this.goToHome}>Cancel</button>
                </div>
                <div className="col-3-6 col-6-6-md col-6-6-xs">

                </div>
              </div>
          </form>
        </div>
      </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  success: state.entry.success
});

NewEntry.propTypes = {
  AddEntry: PropTypes.func,
  success: PropTypes.bool
};

export default connect(mapStateToProps, {
  AddEntry
})(NewEntry);

