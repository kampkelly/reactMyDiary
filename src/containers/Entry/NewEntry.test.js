import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import NewEntry from './NewEntry';
import EntryReducer from '../../reducers/EntryReducer';
import { asyncActions } from '../../util/AsyncUtil';
import { NEW_ENTRY } from '../../actionTypes/EntryConstants';
import { AddEntry } from '../../requests/EntryRequests'
import { entryConstant } from '../../constants/Constants';

const mock = new MockAdapter(axios);
const entry = { title: 'the new title yeah', description: 'the description is now active in this file' };
const mockStore = configureMockStore([thunk]);
const store = mockStore({
  entry: {
    entry: {},
    message: '',
    success: false
  }
});

const props = {
  history: {
    push: jest.fn()
  },
  match: {
    params: {
      id: 1
    }
  },
  addEntry: jest.fn()
};

let component;
let myComponent;

describe('<NewEntry/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <NewEntry {...props} />
      </Provider>
    );
    document.body.innerHTML =
    '<main></main>';
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should show new entry', () => {
    const payload = {};
    const expectedAction = {
      type: 'NEW_ENTRY_SUCCESS',
      payload
    };
    expect(asyncActions(NEW_ENTRY).success(payload)).toEqual(expectedAction);
  });
  it('should dispatch an action when create an entry is loading', () => {
    const payload = false;
    const action = {
      type: 'NEW_ENTRY_LOADING',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      loading: payload
    });
  });
  it('should dispatch an action to create an entry', () => {
    const payload = true;
    const action = {
      type: 'NEW_ENTRY_SUCCESS',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      success: action.payload,
    });
  });
  it('should dispatch an action when creating an entry fails', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    const payload = {
      error: ''
    };
    const action = {
      type: 'NEW_ENTRY_FAILING',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({});
  });
  it('should click submit button', () => {
    document.body.innerHTML =
    '<body>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</body>';
    const input = myComponent.find('input#title');
    input.simulate('change', { target: { value: 'New name' } });
    myComponent.find('button.button-white').simulate('click', {
      preventDefault: () => {
      }
     });
  });
  it('should click cancel button', () => {
    myComponent.find('button.button-cancel').simulate('click');
  });

  it('Create an entry is successful', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    mock.onPost(entryConstant.ENTRIES_URL).reply(201, {
      entry: { id: 1 },
      status: 'Success',
      success: true,
    });

    const mockedActions = [
      {
        type: `${NEW_ENTRY}_LOADING`,
        payload: false,
      },
      {
        type: `${NEW_ENTRY}_SUCCESS`,
        payload: true,
      },
      {
        type: `${NEW_ENTRY}_LOADING`,
        payload: true
      }
    ];

    const store = mockStore({ success: false });
    return store.dispatch(AddEntry(entry.title, entry.description)).then(() => {
      expect(store.getActions()).toEqual(mockedActions);
    });
  });
});
