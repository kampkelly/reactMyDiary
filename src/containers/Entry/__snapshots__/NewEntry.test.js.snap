// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewEntry/> should render without throwing an error 1`] = `
<Connect(NewEntry)
  addEntry={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
/>
`;
